### PR TITLE
xim: use gtk3

### PR DESCRIPTION
**Pull request by Keruspe**
Signed-off-by: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>

### CODE REVIEW

File: client/x11/Makefile.am [hunk 1]
```diff
@@ -40,12 +40,12 @@ ibus_x11_LDADD = \
 	$(libIMdkit) \
 	$(libibus) \
 	@X11_LIBS@ \
-	@GTK2_LIBS@ \
+	@GTK3_LIBS@ \
```
Comment (fujiwarat) [inline client/x11/Makefile.am]: I'm not sure if everyone can accept this change.
Would you able to add if ENABLE_GTK3 ?

if ENABLE_GTK3
ibus_x11_LDADD += @GTK3_LIBS@
endif

if ENABLE_GTK2
ibus_x11_LDADD += @GTK2_LIBS@
endif

File: configure.ac [hunk 2]
```diff
@@ -235,11 +239,7 @@ else
 fi
 
 if test x"$enable_gtk3" = x"yes"; then
-    # check for gtk3, gdk3
-    PKG_CHECK_MODULES(GTK3, [
```
Comment (fujiwarat) [inline configure.ac]: I think the PKG_CHECK_MODULES(GTK3) still makes sense.

File: client/x11/Makefile.am (hunk 1 above)
Comment (Keruspe) [inline client/x11/Makefile.am]: Yes, I can make it conditionnal of course

**Comment by fujiwarat** [comment]
I added some comments on the patch.

**Comment by Keruspe** [comment]
I removed the GDK2 and GDK3 checks which were unused.
I use GTK3 if it is enabled, and GTK2 otherwise

**Comment by fujiwarat** [comment]
Thank you.
I modified the patch slightly and let's move this to the code review.

**Comment by fujiwarat** [comment]
Integrated.
https://github.com/ibus/ibus/commit/4d49a664b9605690a6d0f7da6a0e21c92fb67028

**Comment by fujiwarat** [comment]
I noticed this causes a SEGV in the latest GDK3 on GNOME Wayland.
I will revert the patch or think how gdk-wayland works.

**Pull request closed without merging**